src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -183,13 +209,42 @@ public static PhysicalAddress Parse(string address)
                 validCount++;
             }
 
-            // we too few characters after the last dash
-            if (validCount < 2)
+            // we too few characters after the last delimiter
+            if (validCount < maxSegmentLength)
             {
                 throw new FormatException(SR.Format(SR.net_bad_mac_address, address));
             }
 
             return new PhysicalAddress(buffer);
         }
+
+        private static int GetMaxSegmentLength(string address, char? delimiter)
+        {
+            int segments = 1;
+            int maxSegmentLength = 0;
+            for (int i = 0; i < address.Length; i++)
+            {
+                if (address[i] == delimiter)
+                {
+                    if (maxSegmentLength == 0)
+                    {
+                        maxSegmentLength = i;
```

Comment:
This is only set the first time we see a delimiter.  Is that intentional?  How is that a "max"?

Reply:
Yeah, "max" is kind of poorly named. I updated it to be "validSegmentLength". And yes, I set the segment length on the first segment, then expect all subsequent segments to have the same length.